src/steps/os/unix.rs

```diff
@@ -365,12 +364,48 @@ pub fn run_brew_formula(ctx: &ExecutionContext, variant: BrewVariant) -> Result<
     Ok(())
 }
 
-#[cfg(target_os = "macos")]
+#[cfg(any(target_os = "linux", target_os = "macos"))]
 pub fn run_brew_cask(ctx: &ExecutionContext, variant: BrewVariant) -> Result<()> {
     let binary_name = require(variant.binary_name())?;
-    if variant.is_path() && !BrewVariant::is_macos_custom(binary_name) {
-        return Err(SkipStep(t!("Not a custom brew for macOS").to_string()).into());
+
+    #[cfg(target_os = "macos")]
+    {
+        if variant.is_path() && !BrewVariant::is_macos_custom(binary_name) {
+            return Err(SkipStep(t!("Not a custom brew for macOS").to_string()).into());
+        }
     }
+
+    #[cfg(target_os = "linux")]
+    {
+        // Homebrew cask support was added in version 4.5.0
+        let version_output = Command::new(&binary_name).arg("--version").output_checked_utf8()?;
+
+        let version_line = version_output
+            .stdout
+            .lines()
+            .next()
+            .ok_or_else(|| SkipStep(t!("Failed to get Homebrew version").to_string()))?;
```

Comment:
No silent skipping please! Use `.ok_or_else(|| eyre!(output_changed_message!("brew --version", "version output is empty")))`
Same for below